test/ir_Fujitsu_test.cpp

```diff
@@ -1032,10 +1032,242 @@ TEST(TestIRFujitsuACClass, Timers) {
+  uint16_t rawData[259] = {
+      3220, 1700, 354, 446, 380, 448, 380, 1296, 352, 446, 382, 1296, 354, 448,
+      380, 448, 378, 446, 382, 1296, 352, 1296, 354, 448, 378, 446, 380, 446,
+      382, 1294, 354, 1270, 380, 448, 380, 446, 380, 448, 380, 446, 380, 450,
+      378, 448, 380, 446, 380, 448, 380, 448, 380, 448, 380, 450, 376, 450, 380,
+      448, 378, 1298, 352, 448, 378, 448, 380, 448, 380, 448, 378, 450, 378,
+      450, 378, 448, 378, 1296, 354, 446, 382, 446, 380, 448, 378, 448, 380,
+      1296, 352, 1296, 354, 1296, 354, 1272, 376, 1272, 378, 1296, 354, 1294,
+      354, 1296, 354, 446, 380, 448, 378, 1296, 354, 448, 378, 448, 378, 448,
+      380, 446, 380, 1272, 378, 446, 380, 450, 378, 448, 378, 1296, 354, 1296,
+      354, 446, 380, 448, 378, 448, 378, 446, 382, 446, 380, 1296, 354, 1296,
+      354, 446, 380, 1296, 354, 446, 380, 446, 380, 446, 380, 1294, 354, 448,
+      380, 448, 380, 448, 380, 448, 380, 446, 380, 448, 380, 446, 380, 448,
+      380, 446, 380, 446, 380, 448, 380, 446, 380, 448, 380, 448, 380, 446, 380,
+      1296, 352, 446, 380, 1296, 354, 446, 380, 448, 380, 448, 380, 1296, 354,
+      448, 378, 448, 380, 446, 380, 446, 382, 446, 380, 446, 382, 446, 380,
+      1272, 378, 446, 380, 446, 382, 1294, 354, 446, 382, 1294, 354, 446, 382,
+      446, 382, 446, 380, 448, 380, 448, 380, 448, 380, 448, 378, 1296, 354,
+      446, 382, 446, 380, 1296, 354, 446, 382, 1296, 354, 446, 382, 1294, 354,
+      446, 382, 446, 380, 446, 382};  // UNKNOWN 8383D7DE
+  irsend.sendRaw(rawData, 259, 38);
+  irsend.makeDecodeResult();
+  EXPECT_TRUE(irrecv.decode(&irsend.capture));
+  ASSERT_EQ(FUJITSU_AC, irsend.capture.decode_type);
+  ASSERT_EQ(kFujitsuAcStateLength * 8, irsend.capture.bits);
+  EXPECT_EQ(
+      "Model: 6 (ARREW4E), Id: 0, Power: On, Mode: 4 (Heat), Temp: 19C, "
+      "Fan: 0 (Auto), 10C Heat: Off, Swing: 0 (Off), Command: N/A, "
+      "Outside Quiet: Off, Timer: Off",
+      IRAcUtils::resultAcToString(&irsend.capture));
+  stdAc::state_t r, p;
+  ASSERT_TRUE(IRAcUtils::decodeToState(&irsend.capture, &r, &p));
+}
+
+TEST(TestIRFujitsuACClass, Heat10Deg) {
+  IRFujitsuAC ac(kGpioUnused);
+  const uint8_t heat_on[kFujitsuAcStateLength] = {
+      0x14, 0x63, 0x10, 0x10, 0x10, 0xFE, 0x09, 0x31,
+      0x69, 0x0B, 0x00, 0x23, 0x06, 0x23, 0x20, 0xEF};
+  ac.setRaw(heat_on, kFujitsuAcStateLength);
+  EXPECT_EQ(
+      "Model: 6 (ARREW4E), Id: 1, Power: On, Mode: 3 (Fan), Temp: 21C, "
+      "Fan: 0 (Auto), 10C Heat: On, Swing: 0 (Off), Command: N/A, "
+      "Outside Quiet: Off, Timer: Off",
+      ac.toString());
+  ac.stateReset();
+  ac.setModel(fujitsu_ac_remote_model_t::ARREW4E);
+  ac.setId(1);
+  ac.setMode(kFujitsuAcModeFan);
+  ac.setTemp(21);
+  ac.setFanSpeed(kFujitsuAcFanAuto);
+  ac.setSwing(0);
+  ac.setOutsideQuiet(false);
+  ac.setPower(true);
+  ac.set10CHeat(true);
+  EXPECT_TRUE(ac.get10CHeat());
+  EXPECT_EQ(
+      "Model: 6 (ARREW4E), Id: 1, Power: On, Mode: 3 (Fan), Temp: 21C, "
```

Comment:
Would it be worth testing the other ids here as well?, currently only one bit is tested

Reply:
Not really worth it. The user has tested it already.